src/OrchardCore.Modules/OrchardCore.Workflows/WorkflowPruning/Services/WorkflowPruningManager.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+using OrchardCore.Workflows.Indexes;
+using OrchardCore.Workflows.Models;
+using OrchardCore.Workflows.WorkflowPruning.Models;
+using YesSql;
+using YesSql.Services;
+
+namespace OrchardCore.Workflows.WorkflowPruning.Services;
+
+public class WorkflowPruningManager : IWorkflowPruningManager
+{
+    private readonly ISiteService _siteService;
+    private readonly ISession _session;
+    private readonly IClock _clock;
+
+    public WorkflowPruningManager(ISiteService siteService, ISession session, IClock clock)
+    {
+        _siteService = siteService;
+        _session = session;
+        _clock = clock;
+    }
+
+    public async Task<int> PruneWorkflowInstancesAsync(TimeSpan retentionPeriod)
+    {
+        var dateThreshold = _clock.UtcNow.AddDays(1) - retentionPeriod;
```

Comment:
I know our guy copied this from the audit trail module (because I cleaned up some left over audit trail comments when I pulled it out), so that logic comes directly from there, if its an issue, needs to be fixed in both places.

Reply:
I guess so, but I don't understand why this logic would make sense.